src/lib.rs

```diff
@@ -134,8 +203,14 @@ where
     ///
     /// * `delay_source` - mutable reference to a [DelayUs] provider
     ///
-    pub fn init(&mut self, delay_source: &mut impl DelayUs<u32>) -> Result<(), Error<RST::Error>> {
-        self.madctl = self.model.init(&mut self.di, &mut self.rst, delay_source)?;
+    pub fn init(
+        &mut self,
+        delay_source: &mut impl DelayUs<u32>,
+        options: DisplayOptions,
+    ) -> Result<(), Error<RST::Error>> {
```

Comment:
Need to have a line here that sets `self.orientation`:

```
self.orientation = options.orientation;
```

Otherwise screen dimensions come out wrong. Unless you are call `set_orientation`.